cedar-lean/Cedar/Thm/Data/Map.lean

```diff
@@ -164,31 +372,279 @@ theorem mapOnValues_contains {α β γ} [LT α] [DecidableLT α] [DecidableEq α
   case h_1 h => simp [find?_mapOnValues_some f h]
   case h_2 h => simp [find?_mapOnValues_none f h]
 
+theorem values_mapOnValues [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] {f : β → γ} {m : Map α β} :
+  (m.mapOnValues f).values = m.values.map f
+:= by
+  unfold mapOnValues values kvs
+  induction m.1
+  case nil => simp
+  case cons hd tl ih =>
+    simp only [List.map_cons, List.cons.injEq, true_and]
+    trivial
+
+theorem findOrErr_mapOnValues [LT α] [DecidableLT α] [DecidableEq α] {f : β → γ} {m : Map α β} {k : α} {e : Error} :
+  (m.mapOnValues f).findOrErr k e = (m.findOrErr k e).map f
+:= by
+  unfold findOrErr
+  rw [← find?_mapOnValues]
+  cases m.find? k <;> simp [Except.map]
 
-/-! ### sizeOf -/
+theorem findOrErr_ok_iff_find?_some [LT α] [DecidableLT α] [DecidableEq α] {m : Map α β} {k : α} {v : β} {e : Error} :
+  m.findOrErr k e = .ok v ↔ m.find? k = some v
+:= by
+  unfold findOrErr
+  cases m.find? k <;> simp
 
-theorem sizeOf_lt_of_value [SizeOf α] [SizeOf β] {m : Map α β} {k : α} {v : β}
-  (h : (k, v) ∈ m.1) :
-  sizeOf v < sizeOf m
```

Comment:
Moved the entire `sizeOf` section to the top instead of the bottom, as I needed some `sizeOf` facts elsewhere in this file.